telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala

```diff
@@ -120,4 +121,43 @@ object MetricsTree {
       metricRef.set(Metric.None)
     }
   }
+
+  def flatten(
+    tree: MetricsTree,
+    prefix: String = "",
+    acc: mutable.Buffer[(String, Metric)] = mutable.Buffer()
+  ): Seq[(String, Metric)] = {
+    acc += (prefix -> tree.metric)
+    for ((name, child) <- tree.children) {
+      if (prefix.isEmpty)
+        flatten(child, name, acc)
+      else
+        flatten(child, s"$prefix/$name", acc)
+    }
+    acc
+  }
+
+  /**
+   * Flatten `tree`, skipping empty scopes.
+   */
+  def flattenNones(
```

Comment:
I think it's safe to just skip Metric.None in flatten.  Can't think of a reason why we'd want both behaviors.

Reply:
Yeah, I just wasn't sure, since you'd originally written it that way. Happy to make this the default behaviour